crates/kumod/src/smtp_dispatcher.rs

```diff
@@ -17,7 +17,17 @@ use std::net::SocketAddr;
 #[derive(Deserialize, Serialize, Debug, Clone, Default)]
 pub struct SmtpProtocol {
     #[serde(default)]
-    pub mx_list: Vec<String>,
+    pub mx_list: Vec<MxListEntry>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(untagged)]
+pub enum MxListEntry {
+    // A name that needs to be resolved to its A or AAAA record in DNS,
+    // or an IP domain literal enclosed in square brackets like `[10.0.0.1]`
+    Name(String),
+    // A pre-resolved name and IP address
```

Comment:
Sorry, I should have used the `///` version of this when I proposed this initially!

```suggestion
    /// A name that needs to be resolved to its A or AAAA record in DNS,
    /// or an IP domain literal enclosed in square brackets like `[10.0.0.1]`
    Name(String),
    /// A pre-resolved name and IP address
```